dali/pipeline/data/type_traits.h

```diff
@@ -56,18 +48,14 @@ template <template <typename Backend_> class T, typename Backend>
 struct is_batch_container {
   static constexpr bool value =
       is_backend<Backend>::value &&
-      (is_tensor_vector<T, Backend>::value || is_tensor_list<T, Backend>::value);
+      is_tensor_vector<T, Backend>::value;
 };
 
 template <typename Backend = CPUBackend>
 struct BatchContainer {
```

Comment:
This struct no longer makes much sense, as it's just type alias. Probably to remove at some point?

Reply:
Sure, I can simplify in this PR or in the next one.

Reply:
Removed half the stuff from here and the operator that used it.